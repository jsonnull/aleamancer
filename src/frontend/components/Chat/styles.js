// @flow
import styled from 'styled-components'

export const Container = styled.div`
  background-color: ${props => props.theme.background};
  width: 320px;
  overflow: hidden;
  border-radius: ${props => (props.isPinned ? '0' : '5px')};
  box-shadow: ${props =>
    props.isPinned ? '' : '0 3px 5px 0 rgba(0, 0, 0, 0.2)'};

  height: ${props => (props.isPinned ? '100%' : 'auto')};
  position: ${props => (props.isPinned ? 'relative' : 'absolute')};
  top: ${props => (props.isPinned ? '0' : 'auto')};
  right: ${props => (props.isPinned ? '0' : '1rem')};
  bottom: ${props => (props.isPinned ? '0' : '1rem')};

  display: flex;
  flex-direction: column;
`

export const Loading = styled.div`
  padding: 10px;
`

export const Error = Loading
