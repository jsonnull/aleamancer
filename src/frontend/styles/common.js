// @flow
export const fonts = {
  heading: 'Lora',
  body: `
    /* 1 */ -apple-system, BlinkMacSystemFont,
    /* 2 */ "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans",
    /* 3 */ "Helvetica Neue", sans-serif;
  `
}

export const fontSize = {
  small: '1.3rem',
  normal: '1.5rem',
  medium: '1.8rem',
  large: '2.5rem'
}

export const colors = {
  red: '#bf3542',
  yellow: '#ff974f',
  green: '#8c9b7a',
  blue: '#046380',
  lightBlue: '#057091',
  lightGray: '#e6e6e6'
}

export const timings = {
  easeOutQuad: 'cubic-bezier(0.25, 0.46, 0.45, 0.94)'
}

export const CONSTS = {
  headerHeight: '4.25rem'
}
