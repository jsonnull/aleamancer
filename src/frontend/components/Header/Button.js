// @flow
import styled from 'styled-components'
import { fontSize } from 'frontend/styles/common'

const Button = styled.div`
  height: 3.25rem;
  width: ${props => (props.square ? '3.25rem' : 'auto')};
  line-height: 3.25rem;
  padding: 0 1rem;
  text-align: center;
  font-size: ${fontSize.normal};
  background: ${props => props.theme.backgroundInverted};
  cursor: pointer;
  border-radius: 5px;
  color: ${props => props.theme.colorInverted};

  &:hover {
    background: ${props => props.theme.backgroundInvertedSecondary};
  }
`

export default Button
