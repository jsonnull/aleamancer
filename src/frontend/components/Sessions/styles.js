// @flow
import styled from 'styled-components'
import { fontSize, fonts } from 'frontend/styles/common'

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: stretch;
  flex: 1;
`

export const Body = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
  background: ${props => props.theme.background};
  padding: 0 2rem;
`

export const Heading = styled.h1`
  font-size: ${fontSize.large};
  line-height: 1;
  font-family: ${fonts.heading};
  padding: 1rem 0;
  color: ${props => props.theme.color};
  margin: 0;
`
