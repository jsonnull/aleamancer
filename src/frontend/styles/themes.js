// @flow
import type { ThemeName } from 'common/types'
import { colors } from './common'

export type Theme = {
  name: ThemeName,
  background: string,
  backgroundSecondary: string,
  backgroundInverted: string,
  backgroundInvertedSecondary: string,
  color: string,
  colorSecondary: string,
  colorMedium: string,
  colorInverted: string,
  map: string,
  borderColor: string,
  modalOverlay: string,
  modalBackground: string
}

export const light: Theme = {
  name: 'light',
  background: '#f7f7f7',
  backgroundSecondary: '#ebe3d6',
  backgroundInverted: '#3c3c3c',
  backgroundInvertedSecondary: '#555',
  color: 'black',
  colorSecondary: '#3c3c3c',
  colorMedium: '#999',
  colorInverted: '#f7f7f7',
  map: '#bfbca5',
  borderColor: '#cdc5ba',
  modalBackground: '#f7f7f7',
  modalOverlay: 'rgba(0, 0, 0, 0.5)'
}

export const dark: Theme = {
  name: 'dark',
  background: '#3c3c3c',
  backgroundSecondary: '#2b2b2b',
  backgroundInverted: colors.blue,
  backgroundInvertedSecondary: colors.lightBlue,
  color: '#eee',
  colorSecondary: '#cdc5ba',
  colorMedium: '#999',
  colorInverted: '#eee',
  map: '#1a1a1a',
  borderColor: '#1a1a1a',
  modalBackground: '#2b2b2b',
  modalOverlay: 'rgba(0, 0, 0, 0.7)'
}
