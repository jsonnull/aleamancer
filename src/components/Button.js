// @flow
import styled, { css } from 'styled-components'
import * as themes from 'styles/themes'
import { fontSize, colors } from 'styles/common'

const redMixin = css`
  background-color: ${colors.red};
  color: white;
  &:hover {
    background-color: ${colors.red};
    color: white;
  }
`

const greenMixin = css`
  background-color: ${colors.green};
  color: white;
  &:hover {
    background-color: ${colors.green};
    color: white;
  }
`

const outlineMixin = css`
  background-color: transparent;
  border: 1px solid ${props => props.theme.color};
  color: ${props => props.theme.color};
  &:hover {
    background-color: transparent;
  }
`

export const Button = styled.button`
  display: inline-block;
  font-size: ${fontSize.small};
  line-height: 3rem;
  height: 3rem;
  padding: 0 1rem;
  outline: 0;
  color: ${themes.dark.color};
  background-color: ${colors.blue};
  border: none;
  border-radius: 4px;
  cursor: pointer;
  transition: background, text-shadow;
  transition-duration: 0.1s;
  transition-timing-function: ease-out;
  flex: 1;

  &:hover {
    text-shadow: 0 0 3px white;
    background-color: ${colors.lightBlue};
    transition: background, text-shadow;
    transition-duration: 0.1s;
    transition-timing-function: ease-out;
  }

  ${props => (props.red ? redMixin : null)} ${props =>
      props.green ? greenMixin : null} ${props =>
      props.outline ? outlineMixin : null};
`

export default Button
