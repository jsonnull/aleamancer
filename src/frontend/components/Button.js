// @flow
import styled, { css } from 'styled-components'
import * as themes from 'frontend/styles/themes'
import { colors, fontSize, timings } from 'frontend/styles/common'

function shade(hex: string, lum: number = 0) {
  hex = hex.replace(/[^0-9a-f]/gi, '')

  console.assert(hex.length === 6)

  let rgb = '#'

  for (let i = 0; i < 3; i++) {
    let c = parseInt(hex.substr(i * 2, 2), 16)
    c = Math.round(Math.min(Math.max(0, c + c * lum), 255)).toString(16)
    rgb += ('00' + c).substr(c.length)
  }

  return rgb
}

const boxShadow = color => `inset 0 -20px 10px -10px ${shade(color, -0.06)}`
const hoverBackground = color => shade(color, -0.09)
const hoverBoxShadow = color =>
  `inset 0 -20px 10px -10px ${shade(color, -0.15)};`

const redMixin = css`
  background: ${colors.red};
  box-shadow: ${boxShadow(colors.red)};
  &:hover {
    background: ${hoverBackground(colors.red)};
    box-shadow: ${hoverBoxShadow(colors.red)};
  }
`

const greenMixin = css`
  background: ${colors.green};
  box-shadow: ${boxShadow(colors.green)};
  &:hover {
    background: ${hoverBackground(colors.green)};
    box-shadow: ${hoverBoxShadow(colors.green)};
  }
`

const outlineMixin = css`
  background: transparent;
  box-shadow: none;
  border: 1px solid ${props => props.theme.color};
  color: ${props => props.theme.color};
  &:hover {
    box-shadow: ${props => hoverBoxShadow(props.theme.background)};
    background: transparent;
  }
`

export const Button = styled.button`
  display: inline-block;
  font-size: ${fontSize.small};
  line-height: 3rem;
  height: 3rem;
  padding: 0 1rem;
  outline: 0;
  color: ${themes.dark.color};
  background: ${colors.blue};
  box-shadow: ${boxShadow(colors.blue)};
  border: none;
  border-radius: 4px;
  cursor: pointer;
  transition: background, box-shadow;
  transition-duration: 0.1s;
  transition-timing-function: ${timings.easeOutQuad};
  flex: 1;

  &:hover {
    background: ${hoverBackground(colors.blue)};
    box-shadow: ${hoverBoxShadow(colors.blue)};
    transition: background, box-shadow;
    transition-duration: 0.1s;
    transition-timing-function: ${timings.easeOutQuad};
  }

  ${props => (props.red ? redMixin : null)};
  ${props => (props.green ? greenMixin : null)};
  ${props => (props.outline ? outlineMixin : null)};
`

export default Button
