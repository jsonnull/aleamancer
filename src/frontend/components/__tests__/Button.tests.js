// @flow
import React from 'react'
import { ThemeProvider } from 'styled-components'
import renderer from 'react-test-renderer'
import Button from 'frontend/components/Button.js'
import { light } from 'frontend/styles/themes'

describe('Button component', () => {
  it('renders correctly with no options', () => {
    const tree = renderer.create(<Button />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with red option set', () => {
    const tree = renderer.create(<Button red />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with green option set', () => {
    const tree = renderer.create(<Button green />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with outline option set', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={light}>
          <Button outline />
        </ThemeProvider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
