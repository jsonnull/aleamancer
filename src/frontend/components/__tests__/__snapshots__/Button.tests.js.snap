// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly with green option set 1`] = `
.c0 {
  display: inline-block;
  font-size: 1.3rem;
  line-height: 3rem;
  height: 3rem;
  padding: 0 1rem;
  outline: 0;
  color: #eee;
  background: #046380;
  box-shadow: inset 0 -20px 10px -10px #045d78;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: background,box-shadow;
  transition: background,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background: #8c9b7a;
  box-shadow: inset 0 -20px 10px -10px #849273;
}

.c0:hover {
  background: #045a74;
  box-shadow: inset 0 -20px 10px -10px #03546d;
  -webkit-transition: background,box-shadow;
  transition: background,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
}

.c0:hover {
  background: #7f8d6f;
  box-shadow: inset 0 -20px 10px -10px #778468;
}

<button
  className="c0"
/>
`;

exports[`Button component renders correctly with no options 1`] = `
.c0 {
  display: inline-block;
  font-size: 1.3rem;
  line-height: 3rem;
  height: 3rem;
  padding: 0 1rem;
  outline: 0;
  color: #eee;
  background: #046380;
  box-shadow: inset 0 -20px 10px -10px #045d78;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: background,box-shadow;
  transition: background,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0:hover {
  background: #045a74;
  box-shadow: inset 0 -20px 10px -10px #03546d;
  -webkit-transition: background,box-shadow;
  transition: background,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
}

<button
  className="c0"
/>
`;

exports[`Button component renders correctly with outline option set 1`] = `
.c0 {
  display: inline-block;
  font-size: 1.3rem;
  line-height: 3rem;
  height: 3rem;
  padding: 0 1rem;
  outline: 0;
  color: #eee;
  background: #046380;
  box-shadow: inset 0 -20px 10px -10px #045d78;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: background,box-shadow;
  transition: background,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background: transparent;
  box-shadow: none;
  border: 1px solid black;
  color: black;
}

.c0:hover {
  background: #045a74;
  box-shadow: inset 0 -20px 10px -10px #03546d;
  -webkit-transition: background,box-shadow;
  transition: background,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
}

.c0:hover {
  box-shadow: inset 0 -20px 10px -10px #d2d2d2;
  background: transparent;
}

<button
  className="c0"
/>
`;

exports[`Button component renders correctly with red option set 1`] = `
.c0 {
  display: inline-block;
  font-size: 1.3rem;
  line-height: 3rem;
  height: 3rem;
  padding: 0 1rem;
  outline: 0;
  color: #eee;
  background: #046380;
  box-shadow: inset 0 -20px 10px -10px #045d78;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: background,box-shadow;
  transition: background,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background: #bf3542;
  box-shadow: inset 0 -20px 10px -10px #b4323e;
}

.c0:hover {
  background: #045a74;
  box-shadow: inset 0 -20px 10px -10px #03546d;
  -webkit-transition: background,box-shadow;
  transition: background,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
  transition-timing-function: cubic-bezier(0.25,0.46,0.45,0.94);
}

.c0:hover {
  background: #ae303c;
  box-shadow: inset 0 -20px 10px -10px #a22d38;
}

<button
  className="c0"
/>
`;
